### ISSUE TITLE
[metrics-generator] Add flexibility to exclude externalLabels

### ISSUE THREAD

**Issue by yehaotian** [grafana/tempo#4360]
**Is your feature request related to a problem? Please describe.**
Current metrics generator will add **__metrics_gen_instance** external label with hostname, which will significantly introduce additional cardinality.

```
	hostname, _ := os.Hostname()
	externalLabels["__metrics_gen_instance"] = hostname
```

**Describe the solution you'd like**
Have a config to exclude such external labels to just include needed metrics labels/dimensions.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by javiermolinar** [comment]
Hi, the reason for adding  `__metrics_gen_instance` is to ensure that the series generated by each metrics-generator are unique between them.  I'm not sure if it makes sense to exclude it.

**Comment by javiermolinar** [comment]
Fair enough! We're always open to contributions. If you'd like, feel free to open a pull request so we can explore this feature further

**Comment by yehaotian** [comment]
Hi, would like to reopen this and need some help add a flag to exclude default externalLabels to control metrics cardinality

**Comment by javiermolinar** [comment]
Hi @yehaotian this is not on our roadmap, but PRs are welcome!

**Comment by joe-elliott** [comment]
he created a new issue here: https://github.com/grafana/tempo/issues/5420
